### PR TITLE
solution

### PR DESCRIPTION
**Pull request by YevhenOstrovskiy**
[DEMO LINK](https://YevhenOstrovskiy.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by rtkachuk-moon — APPROVED** [review]
👍

File: src/components/UserInfo/UserInfo.tsx [hunk 1]
```diff
@@ -1 +1,14 @@
-export const UserInfo = () => {};
+import { User } from '../../types/User';
+
+type Props = {
```
Comment (DanilWeda) [inline src/components/UserInfo/UserInfo.tsx:3]:
```suggestion
interface Props = {
```

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1 +1,16 @@
-export const TodoList = () => {};
+import { PreparedTodo } from '../../types/PreparedTodo';
+import { TodoInfo } from '../TodoInfo';
+
+type Props = {
```
Comment (DanilWeda) [inline src/components/TodoList/TodoList.tsx:4]:
```suggestion
interface Props = {
```

File: src/components/AddForm/AddForm.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      return;
+    }
+
+    const reset = () => {
+      setTitle('');
+      setHasTitleError(false);
+
+      setUserId(0);
+      setHasUserIdError(false);
+    };
```
Comment (DanilWeda) [inline src/components/AddForm/AddForm.tsx:43]: good!

File: src/components/AddForm/AddForm.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import { useState } from 'react';
+import usersFromServer from '../../api/users';
+import { PreparedTodo } from '../../types/PreparedTodo';
+import { findUser } from '../../api';
+
+type Props = {
```
Comment (DanilWeda) [inline src/components/AddForm/AddForm.tsx:6]:
```suggestion
interface Props = {
```

File: src/components/AddForm/AddForm.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      setUserId(0);
+      setHasUserIdError(false);
+    };
+
+    onAdd({
+      id: 0,
+      title,
+      completed: false,
+      userId,
+      user: findUser(userId),
```
Comment (DanilWeda) [inline src/components/AddForm/AddForm.tsx:50]: you can create a function that returns you ready todo

File: src/api/index.ts
Comment (DanilWeda) [inline src/api/index.ts:1]: Thanks for splitting

File: src/App.tsx [hunk 6]
```diff
@@ -1,61 +1,38 @@
...
-// import todosFromServer from './api/todos';
+export const initialTodos = getPreparedTodos();
+
+function getNewTodoId(todos: PreparedTodo[]) {
+  const maxId = Math.max(
+    ...todos.map(todo => todo.id),
+  );
+
+  return maxId + 1;
+}
```
Comment (DanilWeda) [inline src/App.tsx:16]: I think it can be a helper function. You don't need to describe it in components.

**Review by DanilWeda — APPROVED** [review]
Great! Approved.
Add some minor comments.  
You can send me a message on the platform when you update the code. 
Many thanks!

**Pull request closed without merging**